kong/plugins/key-auth/handler.lua

```diff
@@ -143,9 +158,14 @@ local function do_authentication(conf)
       key = v
 
       if conf.hide_credentials then
-        query[name] = nil
-        kong.service.request.set_query(query)
-        kong.service.request.clear_header(name)
+        if conf.key_in_query then
+          local raw_query = kong.request.get_raw_query()
+          local new_query = remove_query_key(raw_query, name)
+          kong.service.request.set_raw_query(new_query)
```

Comment:
If an error occurs in the regex gsub, `remove_query_key` will return `nil`, and this call to `set_raw_query` will now throw an error. Let's change `remove_query_key` to return `nil, err` and log the error from here instead. 
Also, I wonder if now we should respond with a 500 in case the query key clearing cannot be done - it can also be seen as a security issue if the plugin config has `hide_credentials` and it has no effect.

Reply:
Fixed in the latest commit. I think returning 500 makes sense but I'm waiting on the final decision you guys make.

Reply:
This is a new error condition; IMO it makes sense to respond with a 500 in that case. Otherwise, the token will be leaked to the upstream service, even though the expectation is it was removed.